packages/@o3r/rules-engine/src/engine/engine.interface.ts

```diff
@@ -74,7 +74,12 @@ export interface EngineRule extends Rule {
 export interface EngineRuleset {
   /** Optional date range where the ruleset will be executed, it supports a dateString or a timestamp as number, more info on javascript Date() documentation */
   validityRange?: { from?: string | number; to?: string | number };
-  /** Component linked to the ruleset, if set it will disable the ruleset execution per default, waiting to a subscription */
+  /** Components linked to the ruleset, if one set, it will disable the ruleset execution per default, waiting to a subscription */
+  linkedComponents?: { library: string; name: string }[];
```

Comment:
So we settle for a solution with a list and support only the OR use case (execution happens whether the component A OR B OR C etc is active). I think @kpanot  was concerned about having to support a AND use case in the future (Execute the rule only if component A AND B are displayed), we have currently no such requirements, but we were discussing the possibility to have an object as a new structure supporting "and" at the root of the object to be able to add a "or" later if the use case comes up without having to do a breaking change in the structure. On my side I have no strong opinion on this, just want to see if everyone is aligned.

Reply:
update done with the new type which includes 'or'